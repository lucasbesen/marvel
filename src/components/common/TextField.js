import React from 'react';
import { TextField as _TextField } from '@material-ui/core';

const TextField = ({ input: { name, onChange, value }, meta, label }) => (
  <_TextField
    name={name}
    label={label}
    helperText={meta.touched ? meta.error : undefined}
    error={meta.error && meta.touched}
    onChange={onChange}
    value={value}
    inputProps={{
      style: {
        width: 250,
      },
    }}
  />
);

export default TextField;
