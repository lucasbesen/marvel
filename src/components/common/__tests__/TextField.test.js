import React from 'react';
import renderer from 'react-test-renderer';

import TextField from '../TextField';

it('should render TextField without error', () => {
  const component = renderer.create(
    <TextField input={{ onBlur: jest.fn(), onChange: jest.fn() }} meta={{}} label="This is a text input" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render TextField with helper texts', () => {
  const component = renderer.create(
    <TextField
      input={{ onBlur: jest.fn(), onChange: jest.fn() }}
      meta={{ touched: true, error: 'Error testing' }}
      label="This is a text input"
    />,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
