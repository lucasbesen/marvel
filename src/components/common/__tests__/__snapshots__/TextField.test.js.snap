// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextField with helper texts 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-error Mui-error"
    data-shrink={false}
  >
    This is a text input
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      style={
        Object {
          "width": 250,
        }
      }
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root Mui-error"
  >
    Error testing
  </p>
</div>
`;

exports[`should render TextField without error 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
  >
    This is a text input
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      style={
        Object {
          "width": 250,
        }
      }
      type="text"
    />
  </div>
</div>
`;
